### PR TITLE
Holy documentation batman

### PR DESCRIPTION
**Pull request by Rickytehduck**
Updates to most non-NPC related command documentation.

A few NPC commands were missed, as well as a few commands which I couldn't get to work/didn't have time to do.

Note that I'm still very inexperienced with GitHub and java so there may be formatting issues with links and other stuff.

### CODE REVIEW

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 1]
```diff
@@ -1659,7 +1690,7 @@ public void registerCoreMembers() {
         // Note: Be careful, gamerules are CASE SENSITIVE.
         //
         // @Tags
-        // TODO: Add tags and then document them!
+        // No tags currently exist
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Open issue/PR for adding tags `<w@world.list_gamerules>` and `<w@world.gamerule[<gamerule>]>` using methods from https://hub.spigotmc.org/javadocs/spigot/org/bukkit/World.html

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 2]
```diff
@@ -1913,11 +1943,10 @@ public void registerCoreMembers() {
         // Does a specified amount of damage usually, but, if no damage is specified, does precisely 1HP worth of damage
         // (half a heart).
         //
-        // TODO: Document Command Details
         //
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: 2 empty comment lines

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 3]
```diff
@@ -2010,13 +2039,17 @@ public void registerCoreMembers() {
         // @Group core
         //
         // @Description
-        // TODO: Document Command Details
+        // Injects a script into the current ScriptQueue.
+        // This means this task will run with all of the queues associated definitions and tags.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: `all of the original queue's definitions and tags`

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 4]
```diff
@@ -2032,13 +2065,19 @@ public void registerCoreMembers() {
         // @Group entity
         //
         // @Description
-        // TODO: Document Command Details
+        // For non-armor stand entities, applies a maximum duration invisibility potion effect. (2,147,483,647t, or about 3.4 years)
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: `Toggles the invisibility state of an entity` is probably sufficient for this description

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 5]
```diff
@@ -2109,28 +2148,31 @@ public void registerCoreMembers() {
...
+        // Note that releasing a mob from a fence post may leave the leash attached to that fence post.
         //
         // @Tags
         // <e@entity.is_leashed>
         // <e@entity.get_leash_holder>
         //
         // @Usage
-        // Use to attach hold an entity in hand.
-        // - leash <npc> holder:<player>
+        // Use to attach a leash to the players target.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: player's

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 6]
```diff
@@ -2109,28 +2148,31 @@ public void registerCoreMembers() {
...
-        // @Short Sticks a leash on target entity, held by a fence post or another entity.
+        // @Short Sticks a leash on target entity, held by a fence or another entity.
         // @Author Alain Blanquet, mcmonkey
         // @Group entity
         //
         // @Description
-        // TODO: Document Command Details
+        // Attaches a leash to the specified entity.
+        // The leash may be attached to a fence, or another entity.
+        // Players/NPCs may not be leashed.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Can non-player type NPCs be leashed?

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 7]
```diff
@@ -2192,7 +2235,7 @@ public void registerCoreMembers() {
         // Kill - Used to detect when a player kills an NPC, player, entity, or player in a specific permission group.
         // Block - Used to detect when a player breaks, places, or collects blocks in the specified list.
         // Item - Used to detect when a player crafts, smelts, or fishes an item on the specified list.
-        // Itemdrop - TODO: Document Command Details
+        // Itemdrop - Used to detect when an item is dropped
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: I would probably just delete this entire command's documentation as it's be deprecated for so long. Leave it registered for ancient script back-support.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 8]
```diff
@@ -2621,7 +2664,7 @@ public void registerCoreMembers() {
         //
         // @Description
         // Add or remove a notable object for that can be used to reference objects such as in events.
-        // TODO: Document Command Details
+        // Notable objects are "permanent" versions of other dObjects. (See: <@link language Particle Effects>)
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: ... particle effects?

Also I would fix the above sentence because it's confusing as hell
`for that can be used to reference objects such as in events` lol

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 9]
```diff
@@ -3070,16 +3113,21 @@ public void registerCoreMembers() {
...
         // @Group entity
         //
         // @Description
-        // TODO: CUBOID OPTION
-        // TODO: Document Command Details
+        // Removes the selected entity. May also take a list of entities to remove.
+        // Any NPC removed this way is completely removed, as if by '/npc remove'.
+        // If a generic entity name is given (see: <@link language entities>)
+        // it will remove all entities of that type from the given world.
+        // Optionally, you may specific a world to target.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: specify

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 10]
```diff
@@ -3311,7 +3359,14 @@ public void registerCoreMembers() {
...
         // @Description
-        // TODO: Document Command Details
+        // Creates, loads , pastes and saves schemactics. Schematics are files containing info about
+        // blocks and the order of those blocks.
+        //
+        // Denizen offers a number of tools to manipulate and work with schematics.
+        // Schematics can be rotated, flipped, pasted with no air, or pasted with a delay.
+        // The "noair" option skips air blocks in the pasted schematics- this means those air blocks will not replace
+        // any blocks in the target location.
+        // The "delayed" option
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: uh

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 11]
```diff
@@ -3521,16 +3576,6 @@ public void registerCoreMembers() {
...
-         * "Cosmos Book":<br>
-         *   Type: Book<br>
-         *   Title: Cosmos, a Personal Voyage<br>
-         *   Author: Carl Sagan<br>
-         *   Text:<br>
-         *   - Every one of us is, in the cosmic perspective, precious. If a human disagrees with<br>
-         *     you, let him live. In a hundred billion galaxies, you will not find another<br>
-         *   - The nitrogen in our DNA, the calcium in our teeth, the iron in our blood, the <br>
-         *     carbon in our apple pies were made in the interiors of collapsing stars. We are <br>
-         *     made of starstuff.<br>
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Why was this removed?

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 12]
```diff
@@ -3564,15 +3609,17 @@ public void registerCoreMembers() {
 
         // <--[command]
         // @Name Scribe
-        // @Syntax scribe [<script>] (<item>/give/equip/{drop <location>})
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Why was `<item>` removed?

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 13]
```diff
@@ -3710,13 +3767,16 @@ public void registerCoreMembers() {
...
-        // TODO: Document Command Details
+        // Makes the linked NPC sit at the specified location.
+        // Use <@link command Stand> to make the NPC stand up again.
         //
         // @Tags
         // None
         //
         // @Usage
-        // TODO: Document Command Details
+        // Make the linked NPC sit at the players cursor location.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: player's

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 14]
```diff
@@ -3908,13 +3974,20 @@ public void registerCoreMembers() {
         // @Group world
         //
         // @Description
-        // TODO: Document Command Details
+        // Changes the state of a lever at the given location.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: I think this also works on doors/trapdoors/etc

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 15]
```diff
@@ -3908,13 +3974,20 @@ public void registerCoreMembers() {
...
+        // Can specify a duration before it returns to the previous state.
+        // By default, will toggle the lever (on to off, or off to on)
         //
         // @Tags
         // <l@location.switched>
         //
         // @Usage
-        // TODO: Document Command Details
+        // At the players location, switch a lever on, no matter what state it was in before.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: `Switch a lever on at the player's location`

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 16]
```diff
@@ -4026,16 +4099,19 @@ public void registerCoreMembers() {
...
         //
         // @Description
         // Changes the current time in a world or the time that a player sees the world in.
-        // TODO: Document Command Details
         // If no world is specified, defaults to the NPCs world. If no NPC is available,
         // defaults to the player's world. If no player is available, an error will be thrown.
+        // If a player is specified, it will change their personal time.
+        // This is separate from the global time, and does not affect other players.
+        // When that player logs off, their time will be reset to the global time.
+        // (May currently be broken)
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: test

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 17]
```diff
@@ -4175,13 +4250,18 @@ public void registerCoreMembers() {
         // @Group npc
         //
         // @Description
-        // TODO: Document Command Details
+        // Toggles whether a NPC can be hurt or not. (Defaults to false)
+        // If you specify a state, sets the NPC to that.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: `Toggles whether an NPC can be hurt. (Defaults to false)` is probably sufficient for this description

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 18]
```diff
@@ -4272,7 +4365,7 @@ public void registerCoreMembers() {
         // @Video /denizen/vids/Loops
         //
         // @Description
-        // TODO: Document Command Details
+        // Loops the commands within the braces so long as the tag returns true.
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Best to just copy the `@Short` description, then explain what *other* parts of this command do (stop/next)

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 19]
```diff
@@ -4256,7 +4339,17 @@ public void registerCoreMembers() {
...
+        // Makes the weather sunny
+        // - weather sunny
+        //
+        // @Usage
+        // Makes the weather storm in world "cookies"
+        // - weather storm world:cookies
+        //
+        // @Usage
+        // Make the weather storm for the player "Rickyling"
+        // - weather type:player storm player:p@Rickyling
```
Comment (mergu) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Best to just use the generic `<player>`

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 20]
```diff
@@ -840,22 +840,28 @@ public void registerCoreMembers() {
...
         // @Author aufdemrand, David Cernat
         // @Group world
         //
         // @Description
-        // TODO: Document Command Details
-        //
+        // Copies a single block to another location.
+        // Does not appear to support most inventories, e.g; Chest, Furnaces, Droppers
+        // (Signs/Spawners/Jukeboxes are supported)
+        // (Does not support cuboids - use '<@link command schematic> command)
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Random stray single quote

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 21]
```diff
@@ -840,22 +840,28 @@ public void registerCoreMembers() {
...
-        // @Short Copies a block to another location, keeping all metadata.
+        // @Short Copies a block to another location, keeping metadata when possible.
         // @Author aufdemrand, David Cernat
         // @Group world
         //
         // @Description
-        // TODO: Document Command Details
-        //
+        // Copies a single block to another location.
+        // Does not appear to support most inventories, e.g; Chest, Furnaces, Droppers
```
[the remark this answers is not in the document]
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Also, official documentation should never say "Does not appear to support [things]". That's the type of text you'd say offhand to someone asking when you don't know much yourself. That's not what an official documentation file says.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 22]
```diff
@@ -1511,7 +1531,16 @@ public void registerCoreMembers() {
         // All the flag values are stored default in "plugins/denizen/saves.yml" file.
         // For an alternative way of storing values, use either yaml (See <@link command yaml>)
         // or sql (See <@link command sql>)
-        // TODO: Document Command Details
+        // Besides being set, flags can be incremented or decremented (added to or subtracted from).
```
[the remark this answers is not in the document]
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: ` incremented or decremented (added to or subtracted from)` is worded like those mean the same thing - they don't.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java (hunk 1 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: A todo that says "add things!" shouldn't be removed and swapped for "We have nothing"... as merg said, open an issue to get that added... or just leave the todo in place.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 23]
```diff
@@ -2010,13 +2039,17 @@ public void registerCoreMembers() {
...
+        // It will also now be part of the queue, so any delays or definitions used in the injected script will be
+        // accessible in the original queue.
         //
         // @Tags
         // None
         //
         // @Usage
-        // TODO: Document Command Details
+        // - inject InjectedTask
+        // Injects the InjectedTask task into the current queue
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: These two lines are out of order - the Usage block is ALWAYS one line "Use to do stuff.." then remaining lines are script code

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java (hunk 4 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: It also does not just apply a potion with duration - I'm not sure all cases, but at least NPC's have it continually retained via an Invisible trait that is added and enabled.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java (hunk 6 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Non-player NPCs are leashable specifically when `/npc leashable` is toggled to allow that - docs should mention that

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java (hunk 7 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Rather for the moment, it should just be left untouched - and a separate PR / plan made regarding removing the entire listener setup

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 24]
```diff
@@ -3070,16 +3113,21 @@ public void registerCoreMembers() {
...
+        // @Syntax remove [<entity>|...] (world:<world name>)
         // @Required 1
         // @Stable stable
-        // @Short Despawns a list of entities, fully removing any NPCs.
+        // @Short Despawns an entity or list of entities, including any NPCs.
         // @Author David Cernat
         // @Group entity
         //
         // @Description
-        // TODO: CUBOID OPTION
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Don't remove a todo specifying to add things without that thing being added

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java (hunk 10 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: This entire change set ... what? You spelled schematic wrong on the first line, have all sorts of formatting issues, and just cut off at the end. What.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java (hunk 11 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: this entire section (`*` prefixed area) needs to be merged into modern docs, not randomly have small portions removed...

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 25]
```diff
@@ -4175,13 +4250,18 @@ public void registerCoreMembers() {
...
+        // If you specify a state, sets the NPC to that.
         //
         // @Tags
         // <n@npc.invulnerable>
         //
         // @Usage
-        // TODO: Document Command Details
+        // Makes an NPC vulnerable.
+        // - vulnerable state:true
+        // Makes an NPC vulnerable if it is not, and invulnerable if it is.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Each usage should start with `@Usage`

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java (hunk 19 above)
Comment (mcmonkey4eva) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Or rather, just ... drop the `player:` entirely and say that it affects the linked player.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 26]
```diff
@@ -1528,19 +1528,17 @@ public void registerCoreMembers() {
         // @Description
         // The flag command sets or modifies custom value storage database entries connected to
         // each player, each NPC, each entity, and the server.
-        // All the flag values are stored default in "plugins/denizen/saves.yml" file.
+        // Flags can have operations performed upon them, such as:
+        // - flag player counter:++            (Adds 1 to the flag)
+        // - flag player counter:+:3           (Adds 3 to the flag)
+        // - flag player counter:-:2           (Remove 2 from the flag)
```
[the remark this answers is not in the document]
Comment (Rickytehduck) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: thank

[the remark this answers is not in the document]
Comment (mergu) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: welc

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 27]
```diff
@@ -1484,13 +1494,23 @@ public void registerCoreMembers() {
         // @Group npc
         //
         // @Description
-        // TODO: Document Command Details
+        // Causes an NPC to begin fishing at the specified location.
+        // Setting catch determines what items the NPC may fish up, and
+        // the chance is the odds of the NPC fishing up an item.
+        // Catch
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Catch!

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 28]
```diff
@@ -1508,10 +1528,20 @@ public void registerCoreMembers() {
...
         // The flag command sets or modifies custom value storage database entries connected to
         // each player, each NPC, each entity, and the server.
-        // All the flag values are stored default in "plugins/denizen/saves.yml" file.
+        // Flags can have operations performed upon them, such as:
+        // Increment a flag by 1:
+        // - flag player counter:++
+        // Increment a flag by 3:
+        // - flag player counter:+:3
+        // Decrement a flag by 2:
+        // - flag player counter:-:2
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Increment and decrement are _not_ the same words as add and subtract. "Increment" _means definitively_ that it goes up by one.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 29]
```diff
@@ -1604,8 +1634,10 @@ public void registerCoreMembers() {
         // @Group core
         //
         // @Description
-        // TODO: Document Command Details
+        // Copies a file from one location to another.
         // The starting directory is server/plugins/Denizen.
+        // May overwrite existing copies of files.
+        // (Does not appear to preserve file extension)
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: "Does not appear to" is not valid documentation

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 30]
```diff
@@ -2027,18 +2059,28 @@ public void registerCoreMembers() {
...
         // @Author aufdemrand, mcmonkey
         // @Group entity
         //
         // @Description
-        // TODO: Document Command Details
+        // For non-armor stand entities, applies a maximum duration invisibility potion.
+        //
+        // For armor stands, toggles them invisible.
+        //
+        // NPCs can't be made invisible if not added to the playerlist.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: The trait automatically adds the playerlist status, which should be made clear here.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 31]
```diff
@@ -3070,16 +3118,21 @@ public void registerCoreMembers() {
 
         // <--[command]
         // @Name Remove
-        // @Syntax remove [<entity>|...]
+        // @Syntax remove [<entity>|...] (world:<world name>)
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: `<world>` not `<world name>`

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 32]
```diff
@@ -3092,23 +3145,23 @@ public void registerCoreMembers() {
...
+        // @Syntax rename [<name>] (npc:<npc>)
         // @Required 1
         // @Stable unstable
         // @Short Renames the linked NPC.
         // @Author aufdemrand
         // @Group npc
         //
         // @Description
-        // TODO: Document Command Details
+        // Renames the linked NPC.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Way too short of a full description.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 33]
```diff
@@ -3092,23 +3145,23 @@ public void registerCoreMembers() {
...
+        // - remove dropped_item world:cookies
         // -->
         registerCoreMember(RemoveCommand.class,
                 "REMOVE", "remove [<entity>|...] (<world>)", 1);
 
 
         // <--[command]
         // @Name Rename
-        // @Syntax rename [<name>]
+        // @Syntax rename [<name>] (npc:<npc>)
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: Not valid syntax doc.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 34]
```diff
@@ -3903,18 +3974,31 @@ public void registerCoreMembers() {
...
+        // @Short Switches state of the block.
         // @Author aufdemrand, Jeebiss, David Cernat
         // @Group world
         //
         // @Description
-        // TODO: Document Command Details
+        // Changes the state of a block at the given location.
+        // Can specify a duration before it returns to the previous state.
+        // By default, will toggle the state (on to off, or off to on).
+        // Works on doors, trapdoors, and gates.
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: It works on _any_ interactable block type, not specifically limited at all.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java (hunk 28 above)
Comment (Rickytehduck) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: mergu gave that example haha, I'll change it.

File: plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java [hunk 35]
```diff
@@ -4272,7 +4388,9 @@ public void registerCoreMembers() {
         // @Video /denizen/vids/Loops
         //
         // @Description
-        // TODO: Document Command Details
+        // Runs a series of braced commands until the tag returns false.
+        // To end a while loop, do - while stop
+        // To jump immediately to the next entry in the loop, do - while next
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: These are written perfectly for placement as `@Usage` sections. Not so great as a description.

Comment (Rickytehduck) [reply plugin/src/main/java/net/aufdemrand/denizen/scripts/commands/BukkitCommandRegistry.java]: While not directly relevant to the current documentation changes, I assume you'll want the current repeat documentation to be later changed too?
(Since this was a modified version from the repeat description)

**Comment by mcmonkey4eva** [comment]
This is not perfect ~~and therefore is being closed~~ but is good enough to really improve the state of command documentation! Thank you for this! ~~Now get back to the documentation mines!~~
